lib/modules.nix

```
@@ -910,6 +910,47 @@ let
     else opt // { type = opt.type.substSubModules opt.options; options = []; };
 
 
+  /*
+    Merge an option's definitions in a way that preserves the priority of the
+    individual attributes in the option value.
+
+    This does not account for all option semantics, such as readOnly.
+
+    Type:
+      option -> attrsOf { highestPrio, value }
+  */
+  mergeAttrDefinitionsWithPrio = opt:
+        let subAttrDefs =
+              lib.concatMap
+                ({ value, ... }@def:
+                  map
+                    (value: def // { inherit value; })
+                    (lib.pushDownProperties value)
```

```suggestion
                    (pushDownProperties value)
```

Already in scope; don't need to use the possibly soon undeprecated version of this name.